### PR TITLE
fix(avatar): state is not updated on avatarUrl change

### LINKED ISSUE #95: (Avatar): Avatar state doesn't update when avatarUrl changes
## Overview
When the avatar component changes from having an image to not having one, the state isn’t reflected properly via the isImageLoaded flag. I experienced this issue specifically in our react-virtualized spaces list from the recents widget, when a new activity triggers the list to be reordered. 

A screenshot is included below with just the avatars part of the recents widget so to not expose my personal space information.
screen shot 2018-07-11 at 4 35 58 pm

## Motivation for or Use Case
The use case is that the react widgets would like to start using collab-ui components, but we cannot until this issue is resolved with our current space list implementation.

## Toolkit Version(s)
7.14.0

## Browsers and Operating System
All browsers on macOS High Sierra, I imagine this is also relevant on other OSes.

## Suggest a Fix 
The fix I suggest is below, to be implemented in `src/lib/Avatar/index.js` in the `render` function:
```javascript
let {
  isImageLoaded
} = this.state;

if (src === null && isImageLoaded) {
  isImageLoaded = false;
}
```
I have already implemented and tested this fix (both the tests in collab-ui-react, and importing it to the widgets repository to ensure it solves the problem), and will open a PR shortly.

### PR DESCRIPTION
**Pull request by moriahmaney**
Would close #95

### CODE REVIEW

File: src/lib/Avatar/index.js [hunk 1]
```diff
@@ -38,9 +38,15 @@ class Avatar extends React.Component {
       icon,
     } = this.props;
     const {
-      isImageLoaded,
       isImageErrored
     } = this.state;
+    let {
```
Comment (shashankcs090) [inline src/lib/Avatar/index.js]: This logic of resetting the state on change of avatar url should not be done here, instead it should be on componentDidUpdate lifecycle method.
You can take this as an example
https://github.com/collab-ui/collab-ui-react/blob/74bed0dcf8d43f23baa96383a04968ac389935ef/src/lib/Input/index.js#L39

**Comment by moriahmaney** [comment]
Updated to address where the logic is handled @shashankcs090  and @pauljeter

File: src/lib/Avatar/index.js [hunk 2]
```diff
@@ -14,6 +14,14 @@ class Avatar extends React.Component {
     isImageErrored: false
   };
 
+  componentDidUpdate() {
+    if (this.props.src === null && this.state.isImageLoaded) {
```
Comment (shashankcs090) [inline src/lib/Avatar/index.js]: I think this should be 
`componentDidUpdate(prevProps){
if(prevProps.src !== this.props.src){
      this.setState({
         isImageLoaded: false,
         isImageErrored: false
      });
}
}`
Basically if the image src of Avatar is reset, it should reset the stale state variables to initial value.

File: src/lib/Avatar/index.js [hunk 3]
```diff
@@ -14,6 +14,15 @@ class Avatar extends React.Component {
     isImageErrored: false
   };
 
+  componentDidUpdate(prevProps) {
+    if(prevProps.src !== this.props.src){
+      this.setState({
+        isImageLoaded: false,
+        isImageErrord: false
```
Comment (bfbiggs) [inline src/lib/Avatar/index.js]: Adjust spelling.

Comment (pauljeter) [inline src/lib/Avatar/index.js]: Misspelled "isImageErrored"

File: src/lib/Avatar/index.js [hunk 4]
```diff
@@ -14,6 +14,15 @@ class Avatar extends React.Component {
     isImageErrored: false
   };
 
+  componentDidUpdate(prevProps) {
+    if(prevProps.src !== this.props.src){
+      this.setState({
```
Comment (bfbiggs) [inline src/lib/Avatar/index.js]: Let me know if the linter isn't catching this for you, but the setState portion should be moved to it's own method.

File: src/lib/Avatar/index.js (hunk 3 above)
Comment (moriahmaney) [reply src/lib/Avatar/index.js]: oops, thanks

File: src/lib/Avatar/index.js (hunk 4 above)
Comment (moriahmaney) [reply src/lib/Avatar/index.js]: good catch, I don't think the contrast for warnings is high enough with my theme right now... Any preference on method name?

**Pull request merged**